### PR TITLE
Changes .load() return when creating new Map object

### PR DESCRIPTION
**Pull request by AngeloGulina**
Based on the suggested implementation, it could be useful to be able to return the new Map instances from the callback. Being able to do something like this:

```javascript
let map = GoogleMapsLoader.load(google => new google.maps.Map( ... ));
```

### CODE REVIEW

**Comment by davidkudera** [comment]
@scriptist good point

@AngeloGulina do you have please any use case for this? Maybe we'll be able to find some better way of your problem. If it is still valid

**Comment by AngeloGulina** [comment]
@scriptist - thanks for the reply. The check is partly on the if/else block. Isn't the library itself assuming to have <code>google</code> in there?

@davidkudera - I'll find the case where I would've needed this and post it. Thanks!

**Comment by davidkudera** [comment]
@AngeloGulina just to clerify things. The only purpose of this package is to easily lazy-load google maps api. So this library don't assume that google api is loaded. It loades google api when it is needed (lazy loading).

So when you first use what you suggest, result will be `null` and the callback will take more time to process (we are loading google maps api here).

It would work for other calls, because google maps api is already loaded. Also callback will be called almost immediatelly.

With your aproach you're trying to mix synchronous and asynchronous programming together.

**Comment by AngeloGulina** [comment]
Sorry @davidkudera, I'm not sure then if I understand well - again I think the <code>if ... else</code> condition there is a safeguard against having <code>null</code> returned. Also, the actual returned value depends on the callback (not even on the <code>google</code> argument). The only fault I can see, in my PL, is it should return the result of the callback/s from more places.

But, again, maybe I got something wrong there. No need do discuss further if you don't see any valid point on this PL. You can close it.

Thank you very much for your reply!

**Comment by davidkudera** [comment]
I can't simply close someones work without proper explanation, can I? :wink: 

So the problem is that your PR will work only sometimes (2nd times +). And just maybe, maybe 3th time, we can't be sure. 

It's the same like if you use jquery ajax. Jquery also can't return result of http request like this: `var result = $.ajax(...)`. Return statement from function is called immediatelly, we are not waiting for anything. But loading something over http takes some time.

```javascript
var function callMe(fn)
{
    var current = true;

    setTimeout(function() {
        current = fn();
    }, 1000);

    return current;
}

var called = callMe(function() {
    return false;
});

console.log(called);

// output: true
```

This is just for simulating our problem. We are returning false, but in reality the `current` variable will be changed after one second. But `return` statement will be called just now with `true` in `current` variable, thus we see `true` in developer console.

---------------------------

Basically if you really need to use such an approach, you should probably load google api synchronously with `<script>` tag in `<head>`. Then google api will be accessible always and everywhere. But you'll also lost precious time loading this library on every page even when you don't use google maps api at all.

This package is purely asynchronous.

---------------------------

Please, do not hesitate to ask any other questions.
